### PR TITLE
Change accepted syntax for code fence

### PR DESCRIPTION
**Pull request by BartoszGrajdek**
We're currently working on adding styles to `code fence` & `inline code` blocks to Live Markdown. After some [discussion](https://swmansion.slack.com/archives/C06BDSWLDPB/p1713274778358479) regarding `code fence`/`pre` block we agreed that the best way to handle it is to limit the accepted syntax variations.

This PR adds these changes directly to ExpensiMark 🙌🏻 

### Fixed Issues
$ [GH_LINK](https://github.com/Expensify/App/issues/39518)

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
All of the tests that have changed in this PR
1. What tests did you perform that validates your changed worked?
I tested it both in Live Markdown & in the Expensify App to make sure that it didn't break anything

# QA
1. What does QA need to do to validate your changes?
Check if there's just one syntax valid for `pre` blocks which should be:
```
(triple backticks)
Content...
(triple backticks)
```
3. What areas to they need to test for regressions?
Mainly Expensify chat & how it treats messages sent with different syntax variations. Live Markdown will include these changes after we get this PR merged 👀 

https://github.com/Expensify/expensify-common/assets/41169160/2f944c36-ae97-4b81-9fe6-b809367fd8cc

### CODE REVIEW

**Review by dangrous — APPROVED** [review]
Cool, this looks good and makes sense as a simplification. Do you mind sharing the discussion (was it an issue? Slack?) that led to this just so I can read that before merging? Thank you!

**Comment by BartoszGrajdek** [comment]
The link to a slack thread is in the issue description, let me know if you have access to it 😄

**Comment by dangrous** [comment]
So I think that link goes to the SWM slack and I can't access. Basically I just wanted to make sure this was run by someone on the Expensify side before I merged? Since it's (sort of) a product decision. I think this makes the most sense to me but wanted to make sure I was the second opinion haha. Thank you!

**Comment by BartoszGrajdek** [comment]
Hi, sorry - I was on sick leave last week 😅 

I've sent you screenshots of our discussion with Expensify on Slack in a DM 🙌🏻

**Comment by dangrous** [comment]
Thank you! That was helpful - merging now.

**Comment by dangrous** [comment]
(and glad you're feeling better!)

File: lib/ExpensiMark.js
Comment (ikevin127) [inline lib/ExpensiMark.js:64]: I think this change might've caused the following issue:
- https://github.com/Expensify/App/issues/46917

Which was fixed in PR https://github.com/Expensify/expensify-common/pull/785, more details in the [proposal](https://github.com/Expensify/App/issues/46917#issuecomment-2272779877).

**Pull request merged**
